tests/typerel/t4799_2.nim

```diff
@@ -0,0 +1,24 @@
+discard """
```

Comment:
Perhaps this bug doesn't deserve so many separate test cases (they slow down the execution of the test suite a little bit).

The alternative would be to use the `accept`/`reject` approach that is frequently used in the rest of the test suite:
https://github.com/nim-lang/Nim/blob/devel/tests/cpp/tcovariancerules.nim#L168